src/app/api/uploadgar/index.js

```diff
@@ -117,20 +118,34 @@ router.post('/uploadgar', upload.single('file'), (req, res, data) => {
       person.documentType = person.documentTypeOther ? 'Other' : person.documentType;
     });
 
-    const crewUpdate = garApi.patch(req.body.garid, 'Draft', { people: crew });
-    const passengerUpdate = garApi.patch(req.body.garid, 'Draft', { people: passengers });
-    const voyageUpdate = garApi.patch(req.body.garid, 'Draft', voyageParser.parse());
-
-    Promise.all([crewUpdate, passengerUpdate, voyageUpdate])
-      .then(() => {
-        res.redirect('/garfile/departure');
-      })
-      .catch((err) => {
-        logger.error('Failed to update API with GAR information');
-        logger.error(err);
-        req.session.failureMsg = 'Failed to read GAR';
-        req.session.failureIdentifier = 'file';
-        return res.redirect('garfile/garupload');
+    createGarApi.createGar(cookie.getUserDbId())
+      .then((apiResponse) => {
+        const parsedResponse = JSON.parse(apiResponse);
+        if (parsedResponse.message) {
+          req.session.failureMsg = 'Failed to create GAR';
+          return req.session.save(() => res.redirect('garfile/garupload'));
+        }
+        logger.info('Created new GAR');
+        const { garId } = parsedResponse;
+        cookie.setGarId(garId);
+        cookie.setGarStatus('Draft');
+
+        const crewUpdate = garApi.patch(garId, 'Draft', { people: crew });
+        const passengerUpdate = garApi.patch(garId, 'Draft', { people: passengers });
+        const voyageUpdate = garApi.patch(garId, 'Draft', voyageParser.parse());
+
+        Promise.all([crewUpdate, passengerUpdate, voyageUpdate])
+          .then(() => {
+            logger.info('Updated GAR with excel data');
+            return req.session.save(() => res.redirect('/garfile/departure'));
+          })
+          .catch((err) => {
+            logger.error('Failed to update API with GAR information');
+            logger.error(err);
+            req.session.failureMsg = 'Failed to read GAR';
+            req.session.failureIdentifier = 'file';
+            return res.redirect('garfile/garupload');
```

Comment:
As a change in functionality, specially because we fixed arrival long and lat, do we need to fix any e2e tests as a result of this? (I do understand though I wont be able to see it as e2e tests repo are not opensourced).

Reply:
In my opinion, no end to end tests need to change as a result of this. E2E tests are testing integration between the various systems, this change only affects the view.